Actually exercise document range tests

apmoriarty opened the pull request:
The test QueryLogicFactory.xml has the collapse uids option set to true. This means that even if the underlying test data contains a uid, that uid will not be persisted as a document range -- effectively turning every document range test into a shard range test. 

All document range tests now explicitly disable the collapse uids option so that document ranges are generated. 

One unit test was ignored because it now returns zero results when using a document range.

apmoriarty committed:
Correctly set the collapseUids option in certain unit tests so that
document range logic is actually exercised. The document uids were
previously ignored because collapsUids was set to true in every case.

Correct ShapesTest index data. Discovered because document ranges were not properly exercised.

TestLimitReturnedGroupsToHitTermGroups exposed an issue with document range queries

hlgp commented:
Documenting what we talked about:  Looks like the ignored test is subject to the InMemoryAccumulo problem.  There are multiple index entries for the term and the UID where it satisfies the query gets clobbered by another.

hlgp committed:
Task/fix commonality tests (#2779)

* Migrate test to mini accumulo cluster and configure shard index aggregators

* Ensure test tables get aggregators applied, correct bunk index entries

* Ensure test tables get aggregators applied, correct bunk index entries

* GitHub Actions: Fix Formatting

* Revert "Migrate test to mini accumulo cluster and configure shard index aggregators"

This reverts commit 484931a618ee082fb7cbddb66f73d316cf0d9862.

Conflicts:
	warehouse/query-core/src/test/java/datawave/query/TestLimitReturnedGroupsToHitTermGroups.java

---------

Co-authored-by: Moriarty <22225248+apmoriarty@users.noreply.github.com>
Co-authored-by: hlgp <hlgp@users.noreply.github.com>
Co-authored-by: GitHub Actions <datawave@github.com>

apmoriarty committed:
Use the correct annotation for JUnit 5

hlgp commented on warehouse/query-core/src/test/java/datawave/query/TestLimitReturnedGroupsToHitTermGroups.java:
```diff
@@ -276,6 +300,7 @@ public void testOneGroup() throws Exception {
         Set<String> goodResults = Sets.newHashSet("CANINE.PET.13:shepherd", "CAT.PET.13:ragdoll", "FISH.PET.13:tetra", "BIRD.PET.13:lovebird",
                         "REPTILE.PET.1:snake", "DOG.WILD.1:coyote", "SIZE.CANINE.3:20,12.5", "SIZE.CANINE.WILD.1:90,26.5");
 
+        // TODO: when executing as a document range no results are found
```
meant to delete this line since it works now

apmoriarty replied:
done

apmoriarty committed:
remove TODO

FineAndDandy commented on warehouse/query-core/src/test/java/datawave/query/IfThisTestFailsThenHitTermsAreBroken.java:
```diff
@@ -131,6 +137,12 @@ public static void beforeAll() throws Exception {
+        @Before
+        public void setup() throws Exception {
+            super.setup();
+            logic.setCollapseUids(true);
```
false?

apmoriarty replied:
Good catch

FineAndDandy commented on warehouse/query-core/src/test/java/datawave/query/util/ShapesIngest.java:
```diff
@@ -134,6 +136,18 @@ public static void writeData(AccumuloClient client, RangeType type) throws Excep
+            String opt = "table.iterator." + scope.name() + ".UIDAggregator.opt.*";
+
+            client.tableOperations().setProperty(SHARD_INDEX, name, "19,datawave.iterators.TotalAggregatingIterator");
+            client.tableOperations().setProperty(SHARD_INDEX, opt, "datawave.ingest.table.aggregator.KeepCountOnlyUidAggregator");
```
this will effectively only produce shard ranges won't it?

apmoriarty replied:
You're thinking of the `KeepCountOnlyNoUidAggregator`

apmoriarty committed:
Correct test setup
